Hadoop Ingestion job configured with hdfs is failing with IllegalArgumentException

Hadoop Batch ingestion job configured with hdfs file system  failing with following exception because API called to copy file cache is expecting directory but what ever the jar trying to copy is a file. Hence getting the excecption.

java.lang.RuntimeException: Caught exception during running - org.apache.pinot.plugin.ingestion.batch.hadoop.HadoopSegmentGenerationJobRunner
	at org.apache.pinot.spi.ingestion.batch.IngestionJobLauncher.kickoffIngestionJob(IngestionJobLauncher.java:152)
	at org.apache.pinot.spi.ingestion.batch.IngestionJobLauncher.runIngestionJob(IngestionJobLauncher.java:121)
	at org.apache.pinot.tools.admin.command.LaunchDataIngestionJobCommand.execute(LaunchDataIngestionJobCommand.java:132)
	at org.apache.pinot.tools.Command.call(Command.java:33)
	at org.apache.pinot.tools.Command.call(Command.java:29)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2045)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2465)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2457)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2419)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
	at picocli.CommandLine.execute(CommandLine.java:2174)
	at org.apache.pinot.tools.admin.command.LaunchDataIngestionJobCommand.main(LaunchDataIngestionJobCommand.java:164)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.apache.hadoop.util.RunJar.run(RunJar.java:328)
	at org.apache.hadoop.util.RunJar.main(RunJar.java:241)
Caused by: java.lang.IllegalArgumentException: /opt/pinot/apache-pinot-1.2.0-bin/plugins-external/pinot-batch-ingestion/pinot-batch-ingestion-hadoop/pinot-batch-ingestion-hadoop-1.2.0-shaded.jar is not a directory
	at org.apache.pinot.plugin.filesystem.HadoopPinotFS.copyFromLocalDir(HadoopPinotFS.java:225)
	at org.apache.pinot.spi.filesystem.NoClosePinotFS.copyFromLocalDir(NoClosePinotFS.java:104)
	at org.apache.pinot.plugin.ingestion.batch.hadoop.HadoopSegmentGenerationJobRunner.addMapperJarToDistributedCache(HadoopSegmentGenerationJobRunner.java:352)
	at org.apache.pinot.plugin.ingestion.batch.hadoop.HadoopSegmentGenerationJobRunner.run(HadoopSegmentGenerationJobRunner.java:226)
	at org.apache.pinot.spi.ingestion.batch.IngestionJobLauncher.kickoffIngestionJob(IngestionJobLauncher.java:150)
	... 19 more

--- a/pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-hadoop/src/main/java/org/apache/pinot/plugin/ingestion/batch/hadoop/HadoopSegmentGenerationJobRunner.java
+++ b/pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-hadoop/src/main/java/org/apache/pinot/plugin/ingestion/batch/hadoop/HadoopSegmentGenerationJobRunner.java
@@ -349,7 +349,11 @@ public class HadoopSegmentGenerationJobRunner extends Configured implements Inge
       throws Exception {
     File ourJar = new File(getClass().getProtectionDomain().getCodeSource().getLocation().toURI());
     Path distributedCacheJar = new Path(stagingDirURI.toString(), ourJar.getName());
-    outputDirFS.copyFromLocalDir(ourJar, distributedCacheJar.toUri());
+    if (ourJar.isDirectory()) {
+      outputDirFS.copyFromLocalDir(ourJar, distributedCacheJar.toUri());
+    } else {
+      outputDirFS.copyFromLocalFile(ourJar, distributedCacheJar.toUri());
+    }
     job.addFileToClassPath(distributedCacheJar);
   }
 

